crates/ty_project/src/db.rs

```diff
@@ -64,18 +66,28 @@ impl ProjectDatabase {
             system: Arc::new(system),
         };
 
+        // Load the database from the persistent cache.
+        //
         // TODO: Use the `program_settings` to compute the key for the database's persistent
         //   cache and load the cache if it exists.
```

Comment:
@MichaReiser I'm not actually sure if we need a cache key, because the `Project` input is deserialized and updated if the settings have changed?

Reply:
A not uncommon use case is to check a project with different Python versions: Each run then uses a different `ProgramSettings`. Ideally, the caches would be separate